jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -102,6 +105,38 @@
         appRoot, rawConfiguration.getJvmFlags(), mainClass);
   }
 
+  /**
+   * Checks whether or not current Java version is compatible with the base image.
+   *
+   * @param baseImageConfiguration the configured base image
+   * @param fromImageParameterName the name of the base image configuration parameter
+   * @throws InvalidImageReferenceException if the project is incompatible with the base image
+   */
+  public static void checkJavaVersion(
+      @Nullable String baseImageConfiguration, String fromImageParameterName)
+      throws InvalidImageReferenceException {
+    // TODO: check for other Java 8 base images?
+    if (baseImageConfiguration == null
```

Comment:
nit: I would invert the condition here and early-return so the rest of the method is not inside this conditional